File: server/app/models/session.py
from datetime import datetime
from typing import Optional

from .camel_model import CamelModel
from .user import User
from .recommendation import Recommendation
from .song import Playlist


class SessionCore(CamelModel):
    id: Optional[str] = None
    name: str
    host_id: Optional[str] = None
    host_name: Optional[str] = None
    guests: dict[str, User] = {}
    invite_link: Optional[str] = None
    creation_date: Optional[datetime] = None
    voting_start_date: Optional[datetime] = None
    voting_end_date: Optional[datetime] = None


class Session(SessionCore):
    playlist: Playlist
    recommendations: list[Recommendation] = []
    is_running: bool = True

File: client/src/types/Session.ts
import { User } from './User';
import { Playlist } from '@/types/Playlist';
import { Song } from '@/types/Song';

export class Session {
    id: string
    name: string
    hostId: string
    hostName: string
    inviteLink: string
    guests: {[key: string]: User}
    playlist: Playlist
    creationDate: Date
    recommendations: Song[]

    constructor(data: Session) {
        this.id = data.id
        this.name = data.name
        this.hostId = data.hostId
        this.hostName = data.hostName
        this.inviteLink = data.inviteLink
        this.guests = data.guests
        this.playlist = data.playlist
        this.creationDate = data.creationDate
        this.recommendations = data.recommendations
    }
}
